test_util/test_azure.py

```diff
@@ -0,0 +1,139 @@
+        AssertionError: if any environment variables or resources are missing
+            or do not conform
+    """
+    options = type('Options', (object,), {})()
+
+    # Required
+    options.public_ssh_key = calculate_environment_variable('AZURE_PUBLIC_SSH_KEY')
+    options.subscription_id = calculate_environment_variable('AZURE_SUBSCRIPTION_ID')
+    options.tenant_id = calculate_environment_variable('AZURE_TENANT_ID')
+    options.client_id = calculate_environment_variable('AZURE_CLIENT_ID')
+    options.client_secret = calculate_environment_variable('AZURE_CLIENT_SECRET')
+    options.template_url = calculate_environment_variable('DCOS_TEMPLATE_URL')
+
+    # Provided if not set
+    options.name = os.getenv('DCOS_NAME', 'testing-{}'.format(random_id(10)))
+    options.ssh_key_path = os.getenv('DCOS_SSH_KEY_PATH', 'ssh_key')
+    options.location = os.getenv('AZURE_LOCATION', 'East US')
+    options.linux_user = os.getenv('AZURE_LINUX_USER', 'dcos')
+    # Prefixes must not begin with a number
+    options.agent_prefix = os.getenv('AZURE_AGENT_PREFIX', 'test' + random_id(10).lower())
+    options.master_prefix = os.getenv('AZURE_MASTER_PREFIX', 'test' + random_id(10).lower())
+    options.vm_size = os.getenv('AZURE_VM_SIZE', 'Standard_D2')
+    options.num_agents = int(os.getenv('AGENTS', '2'))
+    options.name_suffix = os.getenv('AZURE_DCOS_SUFFIX', '12345')
+    options.oauth_enabled = os.getenv('AZURE_OAUTH_ENABLED', 'false') == 'true'
+    options.vm_diagnostics_enabled = os.getenv('AZURE_VM_DIAGNOSTICS_ENABLED', 'true') == 'true'
+    options.azure_cleanup = os.getenv('AZURE_CLEANUP', 'true') == 'true'
+    options.ci_flags = os.getenv('CI_FLAGS', '')
+
+    add_env = {}
+    prefix = 'TEST_ADD_ENV_'
+    for k, v in os.environ.items():
+        if k.startswith(prefix):
+            add_env[k.replace(prefix, '')] = v
+    options.add_env = add_env
+    options.pytest_cmd = os.getenv('DCOS_PYTEST_CMD', "py.test -rs -vv -m 'not ccm' ") + os.getenv('CI_FLAGS', '')
+    return options
+
+
+def main():
+    options = check_environment()
+    aw = AzureWrapper(
+        options.subscription_id,
+        options.tenant_id,
+        options.client_id,
+        options.client_secret)
+    arm_deploy = TemplateDeployment.create(
+        azure_wrapper=aw,
+        template_uri=options.template_url,
+        location=options.location,
+        name=options.name,
+        public_key=options.public_ssh_key,
+        master_prefix=options.master_prefix,
+        agent_prefix=options.agent_prefix,
+        admin_name=options.linux_user,
+        oauth_enabled=options.oauth_enabled,
+        vm_size=options.vm_size,
+        agent_count=options.num_agents,
+        name_suffix=options.name_suffix,
+        vm_diagnostics_enabled=options.vm_diagnostics_enabled,
+        rollback=options.azure_cleanup)
+    result = 1
+    try:
+        with SSHTunnel(options.linux_user, options.ssh_key_path, arm_deploy.outputs['masterFQDN'], port=2200) as t:
```

Comment:
Can we add a `from_azure` like https://github.com/dcos/dcos/blob/master/test_util/cluster.py#L169 then call https://github.com/dcos/dcos/blob/master/test_util/cluster.py#L472